lib/ansible/module_utils/network/nos/nos.py

```diff
@@ -0,0 +1,148 @@
+#
+# (c) 2018 Extreme Networks Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+import json
+from ansible.module_utils._text import to_text
+from ansible.module_utils.network.common.utils import to_list, ComplexList
+from ansible.module_utils.connection import Connection
+
+
+def get_connection(module):
+    """Get switch connection
+
+    Creates reusable SSH connection to the switch described in a given module.
+
+    Args:
+        module: A valid AnsibleModule instance.
+
+    Returns:
+        An instance of `ansible.module_utils.connection.Connection` with a
+        connection to the switch described in the provided module.
+
+    Raises:
+        AnsibleConnectionFailure: An error occurred connecting to the device
+    """
+    if hasattr(module, 'nos_connection'):
+        return module.nos_connection
+
+    capabilities = get_capabilities(module)
+    network_api = capabilities.get('network_api')
+    if network_api == 'cliconf':
+        module.nos_connection = Connection(module._socket_path)
+    else:
+        module.fail_json(msg='Invalid connection type %s' % network_api)
+
+    return module.nos_connection
+
+
+def get_capabilities(module):
+    """Get switch capabilities
+
+    Collects and returns a python object with the switch capabilities.
+
+    Args:
+        module: A valid AnsibleModule instance.
+
+    Returns:
+        A dictionary containing the switch capabilities.
+    """
+    if hasattr(module, 'nos_capabilities'):
+        return module.nos_capabilities
+
+    capabilities = Connection(module._socket_path).get_capabilities()
```

Comment:
Calling method on connection object can result in ConnectionError exception which needs to handled in module code else it will result in stacktrace in output. Please refer [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/network/ios/ios.py#L87)

Reply:
Thanks @ganeshrn - I think my code is branched from the slxos modules, which originally branched from the IOS modules before those improvements went in. I'll update this - and looks like a future PR needed to port the changes to the SLXOS modules too